### ISSUE TITLE
Running in CI pipeline fails because plugin is loading

### ISSUE THREAD

**Issue by Giuspepe** [invertase/dart_custom_lint#16]
My custom linter works fine in the IDE and when run on the terminal with `flutter pub run custom_lint`. 
But when I try to run `flutter analyze` in a CI pipeline to analyze my code for the default lints, I get this output and the command fails:

```
Analyzing my_app...                                        

warning • The plugin is currently starting • pubspec.yaml:121:2 • custom_lint_plugin_loading

1 issue found. (ran in 101.0s)
```

I've tried adding another CI step before the `flutter analyze` step which executes the custom linter first. The `flutter pub run custom_lint` CI step works fine, but even after that step finished I still get the above error when running `flutter analyze` in subsequent CI step.

Is there a way to wait until the plugin has started? Or is there some workaround so I can use `custom_lint` in a CI pipeline?

**Comment by jscti** [comment]
+1 this and unfortunately the only option is to disable all this lib as it is linked to the dart analyzer (CI and IDE)

I think the `if` from `packages/custom_lint/lib/src/analyzer_plugin/lints.dart` should be totally deleted (message not very helpful anyway) : 
![image](https://user-images.githubusercontent.com/1594035/180749621-1224ee13-8985-4ed4-b196-533c7b1bd99a.png)

Great lib BTW ;)

**Comment by Giuspepe** [comment]
I'm currently using the following workaround on CI:

1. run `flutter pub run custom_lint`
2. run a script which removes `custom_lint` from `analysis_options.yaml`
3. run `flutter analyze`

<details>
  <summary>Python script for editing analysis_options.yaml </summary>

```python
import yaml

with open('analysis_options.yaml', 'r+') as file:
    pubspec = yaml.safe_load(file)
    file.seek(0)
    file.truncate()
    del pubspec["analyzer"]["plugins"]
    yaml.safe_dump(pubspec, file)

```
</details>

This way, you can use both `flutter pub run custom_lint` as well as `flutter analyze` on CI without `flutter analyze` causing the `custom_lint_plugin_loading` error.

**Comment by victor-tinoco** [comment]
Nice workaround @Giuspepe.

I had the same issue here, and following your way I did the same in the shell script, which seems simpler, and independent of having python in the environment.

I will leave it here, hope to be useful to someone that is facing the same issue:

```sh
echo 'Removing custom lint from the yaml file...'

sed -i '' '/analyzer:/d' $1
sed -i '' '/plugins:/d' $1
sed -i '' '/- custom_lint/d' $1

exit 0
```

So, just run:
```sh
sh remove_custom_lint.sh ./path/to/analysis_options.yaml
```

Note: in my specific case, where we work with multiple modules using this custom rule, we had to receive dynamically the _yaml_ file path as a param, but if you are in a simple scenario it is quite simple to just use `$0` to build this path.
